lib/cluster-providers/mng-cluster-provider.ts

```diff
@@ -55,6 +55,16 @@ export interface MngClusterProviderProps extends eks.CommonClusterOptions {
      */
     amiReleaseVersion?: string;
 
+    /**
+     * The custom AMI for the node group. `amiType` and `amiReleaseVersion` will be ignored if this is set.
+     */
+    customAmi?: IMachineImage;
+
+    /**
+     * The userData for worker node when using custom AMI. Only applicable when customAmi is used.
+     */
+    userData?: UserData;
```

Comment:
This is confusing as on the surface it appears to be unrelated to the custom AMI. 
It would be easier to comprehend if you use a nested structure like 
```
customAmi?: {
    machineImage: IMachineImage,
    userData?: UserData
}
```

Reply:
Fixed. Good suggestion.